tests/integration/components/dialog-test.js

```diff
@@ -233,7 +234,50 @@ module('Integration | Component | <Dialog>', function (hooks) {
         async function () {}
       );
 
-      todo('it should add a scroll lock to the html tag', async function () {});
+      test('it should add a scroll lock to the html tag', async function (assert) {
+        const portalRoot = getPortalRoot();
+        this.set('isOpen', false);
+
+        await render(hbs`
+          <button id="trigger" type="button" {{on "click" (set this "isOpen" true)}}>
+            Trigger
+          </button>
+          <Dialog
+            class="relative bg-blue-500"
+            @isOpen={{this.isOpen}}
+            @onClose={{set this "isOpen" false}}
+            as |d|
+          >
+            <d.Overlay data-test-overlay>Hello</d.Overlay>
+            <div tabindex="0"></div>
+          </Dialog>
+        `);
+
+        assert
+          .dom(portalRoot)
+          .doesNotHaveStyle(
+            { overflow: 'hidden' },
+            'The page is not initially "locked"'
+          );
+
+        await click(document.getElementById('trigger'));
+
+        assert
+          .dom(portalRoot)
+          .hasStyle(
+            { overflow: 'hidden' },
+            'The page becomes "locked" when the dialog is open'
+          );
+
+        await click('[data-test-overlay]');
+
+        assert
+          .dom(portalRoot)
+          .doesNotHaveStyle(
+            { overflow: 'hidden' },
+            'The page is "unlocked" when the dialog closes'
+          );
```

Comment:
Technically this assertion is "extra"; the upstream tests for the React implementation don't test cleaning things up after the `Dialog` is closed, but that seemed like a useful thing for us to have!

Reply:
I'm not sure why, but this test seems to fail when I run it locally:

<img width="892" alt="Screenshot 2021-11-20 at 08 13 00" src="https://user-images.githubusercontent.com/2526/142719465-e6feb9b8-e174-457e-ba8a-5f96f2cba4a9.png">

Does it pass for you?

Reply:
Yup, it's passing locally for me

For some reason a bunch of the `Listbox` tests fail, but the `Dialog` ones all pass

Reply:
Weird, thanks. I'll try on another machine